atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/TracingKeyValueService.java

```diff
@@ -69,10 +69,10 @@ protected KeyValueService delegate() {
     }
 
     @Override
-    public void addGarbageCollectionSentinelValues(TableReference tableRef, Set<Cell> cells) {
+    public void addGarbageCollectionSentinelValues(TableReference tableRef, Iterable<Cell> cells) {
         //noinspection unused - try-with-resources closes trace
         try (CloseableTrace trace = startLocalTrace("addGarbageCollectionSentinelValues({}, {} cells)",
-                tableRef, cells.size())) {
+                tableRef, Iterables.size(cells))) {
```

Comment:
I mentioned this problem in the "Concerns" section of the PR description. Other methods in this class already call Iterable.size(), that's why I stuck with it.

@gsheasby Using a Collection interface doesn't guarantee that size() is O(1). For example, see Collections2.filter(). And vice versa: passing an Iterable doesn't automatically mean O(N) - take a look at the impl of the Iterables.size().

Ideally, we want to somehow refactor this class to not call size() at all if tracing is disabled, but I'd rather do that in a separate PR because, again, the problem is not limited to this particular method. The other option is to not log the size, but that would make this tracing much less useful.